### ISSUE TITLE
"media parameter is invalid" when uploading large image

### ISSUE THREAD

**Issue by gusrub** [baedert/corebird#455]
Trying to upload a 3.2Mb PNG image and getting the following:

`44: media parameter is invalid.(ComposeTweetWindow.vala:272)` 

Resizing the image to JPEG (which lowered the size to around 400Kb) allowed me to tweet the message.

Reading the [twitter REST api docs](https://dev.twitter.com/rest/public/uploading-media#keepinmind) it seems the limit is 5Mb.

Not sure if the media limit under settings affects this, I bumped it up to 4 megs, no joy.

**Comment by baedert** [comment]
Yeah, no clue why that would happen. The image gets uploaded successfully, the ID is fine. Seems to be a problem on Twitter's side.

**Comment by gusrub** [comment]
Uhmm ok I was lazy to debug myself yesterday, maybe I'll check that out on the weekend. Thanks.

**Comment by baedert** [comment]
FWIW, the documentation here got [updated](https://twittercommunity.com/t/getting-media-parameter-is-invalid-after-successfully-uploading-media/58354/5):
> The maximum image size for this endpoint is 5MB. Depending on how the image is used, further limits apply. For example, if the image is attached to a Tweet, the maximum limit in this case is 3MB.

So a 3MB image can't work, but I can also reproduce the issue with a slightly-over-2MB file.

**Comment by gusrub** [comment]
Yup, maybe catching that situation before uploading the file, getting its file size and if it's less than 3MB then allow the upload, otherwise display a warning stating that the attachment should be smaller.
